morph_tool/converter.py

```diff
@@ -155,21 +158,53 @@ def cylinder_to_cylindrical_contour(neuron):
     neuron.soma.type = SomaType.SOMA_SIMPLE_CONTOUR
 
 
-def single_point_sphere_to_circular_contour(neuron):
+def single_point_sphere_to_circular_contour(neuron, ensure_NRN_area=False):
     """Single point soma to contour soma.
 
     Transform a single point soma that represents a sphere into a circular contour that represents
     the same sphere.
+
+    With ensure_NRN_area=True, we perform a search on radius to ensure an equivalent
+    soma surface area, as measured by NEURON with morph_tool.neuron_surface.get_NEURON_surface.
     """
     L.info('Converting 1-point soma (spherical soma) to circular contour '
            'representing the same sphere')
     assert neuron.soma_type == SomaType.SOMA_SINGLE_POINT
-    radius = neuron.soma.diameters[0] / 2.
-    points, diameters = _create_contour(radius)
-    origin = neuron.soma.points[0]
-    neuron.soma.points = points + origin
-    neuron.soma.diameters = diameters
-    neuron.soma.type = SomaType.SOMA_SIMPLE_CONTOUR
+
+    swc_radius = neuron.soma.diameters[0] / 2.0
+
+    def make_soma(radius):
+        points, diameters = _create_contour(radius)
+        origin = neuron.soma.points[0]
+        neuron.soma.points = points + origin
+        neuron.soma.diameters = diameters
+        neuron.soma.type = SomaType.SOMA_SIMPLE_CONTOUR
+
+    make_soma(swc_radius)
+
+    if ensure_NRN_area:
+
+        # pylint: disable=import-outside-toplevel
+        from morph_tool.neuron_surface import get_NEURON_surface
+
+        surf = 4.0 * np.pi * swc_radius**2
+        filename = f"{str(uuid.uuid4())}.asc"
```

Comment:
I think it would be better to use a tempfile or a StringIO if Neuron is able to read it in `get_NEURON_surface`.

Reply:
yes, I wasn't sure how to best do this

Reply:
You could try this: https://docs.python.org/3/library/tempfile.html#tempfile.SpooledTemporaryFile

Reply:
it does not work with morphio:

```
morphio._morphio.UnknownFileType: Filename: <tempfile.SpooledTemporaryFile object at 0x7ff5dac9ba90> must have one of the following extensions: swc, asc or h5
```